File: Ruby/Smallest_Multiple.rb
# 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
#
# What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?


def smallest_mult(range)
  result = range.last
  no_remainder = false

  until no_remainder
    mods_all = true
    (range.first..range.last).each do |i|
      if result % i > 0
        mods_all = false
        break
      end
    end
    mods_all ? no_remainder = true : result += range.last
  end

  result
end

File: Javascript/Smallest_Multiple.js
// 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
//
// What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?


function smallest_mult(range) {
  var result = range[1]
  var no_remainder = false

  while (!no_remainder) {
    var mods_all = true
    for (var i = range[0]; i <= range[1]; i++) {
      if (result % i > 0) {
        mods_all = false
        break
      }
    }
    mods_all ? no_remainder = true : result += range[1]
  }
  return result
}
